worlds/aquaria/Regions.py

```diff
@@ -0,0 +1,1363 @@
+    """`player` in `state` has the Sun crystal item"""
+    return state.has("Has sun crystal", player)
+
+
+def _has_li(state, player: int) -> bool:
+    """`player` in `state` has Li in it's team"""
+    return state.has("Li and Li song", player)
+
+
+def _has_damaging_item(state, player: int) -> bool:
+    """`player` in `state` has the shield song item"""
+    return (state.has("Energy form", player) or
+            state.has("Nature form", player) or
+            state.has("Beast form", player) or
+            state.has("Li and Li song", player) or
+            state.has("Baby nautilus", player) or
+            state.has("Baby piranha", player) or
+            state.has("Baby blaster", player)
+            )
+
+
+def _has_shield_song(state, player: int) -> bool:
+    """`player` in `state` has the shield song item"""
+    return state.has("Shield song", player)
+
+
+def _has_bind_song(state, player: int) -> bool:
+    """`player` in `state` has the bind song item"""
+    return state.has("Bind song", player)
+
+
+def _has_energy_form(state, player: int) -> bool:
+    """`player` in `state` has the energy form item"""
+    return state.has("Energy form", player)
+
+
+def _has_beast_form(state, player: int) -> bool:
+    """`player` in `state` has the beast form item"""
+    return state.has("Beast form", player)
+
+
+def _has_nature_form(state, player: int) -> bool:
+    """`player` in `state` has the nature form item"""
+    return state.has("Nature form", player)
+
+
+def _has_sun_form(state, player: int) -> bool:
+    """`player` in `state` has the sun form item"""
+    return state.has("Sun form", player)
+
+
+def _has_light(state, player: int) -> bool:
+    """`player` in `state` has the light item"""
+    return state.has("Baby dumbo", player) or _has_sun_form(state, player)
+
+
+def _has_dual_form(state, player: int) -> bool:
+    """`player` in `state` has the dual form item"""
+    return _has_li(state, player) and state.has("Dual form", player)
+
+
+def _has_fish_form(state, player: int) -> bool:
+    """`player` in `state` has the fish form item"""
+    return state.has("Fish form", player)
+
+
+def _has_spirit_form(state, player: int) -> bool:
+    """`player` in `state` has the spirit form item"""
+    return state.has("Spirit form", player)
+
+
+def _has_big_bosses(state, player: int) -> bool:
+    """`player` in `state` has beated every big bosses"""
+    return state.has_all("Fallen God beated", "Mithalan God beated", "Drunian God beated",
+                         "Sun God beated", "The Golem beated", player)
+
+
+def _has_mini_bosses(state, player: int) -> bool:
+    """`player` in `state` has beated every big bosses"""
+    return state.has_all("Nautilus Prime beated", "Blaster Peg Prime beated", "Mergog beated",
+                         "Mithalan priests beated", "Octopus Prime beated", "Crabbius Maximus beated",
+                         "Mantis Shrimp Prime beated", "King Jellyfish God Prime beated", player)
```

Comment:
```suggestion
def _has_mini_bosses(state, player: int) -> bool:
    """`player` in `state` has beated every big bosses"""
    return state.has_all({"Nautilus Prime beated", "Blaster Peg Prime beated", "Mergog beated",
                         "Mithalan priests beated", "Octopus Prime beated", "Crabbius Maximus beated",
                         "Mantis Shrimp Prime beated", "King Jellyfish God Prime beated"}, player)
```

Missed the {}

Reply:
Hmmm? Ok. Yes. But how does that actually worked in my case. I launch the generate and the tests and it all works. I will do the modifications, but this a mystery.

Reply:
That's very bizarre that it didn't error out or anything, will look into it a little